build/release/Makefile

```
@@ -120,8 +120,13 @@ HELM_URL := $(HELM_BASE_URL)/$(HELM_CHANNEL)
 promote.helm: $(HELM)
 #	copy existing charts to a temp dir, then combine with new charts, reindex, and upload
 	$(S3_SYNC) s3://$(HELM_S3_BUCKET)/$(HELM_CHANNEL) $(HELM_TEMP)
+	# delete unnecessary files:
+	rm -fr $(HELM_TEMP)/*tgz
+	# rename current index to distinguish the previous file.
+	mv $(HELM_TEMP)/index.yaml $(HELM_TEMP)/previous_index.yaml
 	$(S3_SYNC) s3://$(S3_BUCKET)/build/$(BRANCH_NAME)/$(VERSION)/charts $(HELM_TEMP)
-	$(HELM) repo index --url $(HELM_URL) --merge $(HELM_TEMP)/index.yaml $(HELM_TEMP)
+	$(HELM) repo index --url $(HELM_URL) --merge $(HELM_TEMP)/previous_index.yaml $(HELM_TEMP)
+	rm -f $(HELM_TEMP)/previous_index.yaml
 	$(S3_SYNC_DEL) $(HELM_TEMP) s3://$(HELM_S3_BUCKET)/$(HELM_CHANNEL)
```

We need to not pass the `--delete` flag, or else all the previous `.tgz files will be removed from the s3 bucket.
```suggestion
	$(S3_SYNC) $(HELM_TEMP) s3://$(HELM_S3_BUCKET)/$(HELM_CHANNEL)
```